test/Makefile

```diff
@@ -291,14 +296,30 @@ test-error: test_error
         fi; \
         echo "Error check test passed"
 
-test-special: test-int test-highd test-sixd
+test-zfparr: test_write_lib
+	@./test_write_lib zfparr=1 rate=10 2>&1 1>/dev/null; \
+	if [[ $$? -ne 0 ]]; then \
+	    echo "ZFP array direct write test failed"; \
+	    exit 1; \
+	fi; \
+	env LD_LIBRARY_PATH=$(HDF5_LIB):$(LD_LIBRARY_PATH) HDF5_PLUGIN_PATH=$(H5Z_ZFP_PLUGIN) $(HDF5_BIN)/h5diff -v -d 0.01 test_zfp.h5 test_zfp.h5 zfparr_original zfparr_direct 2>&1 1>/dev/null; \
```

Comment:
We should compare data written with compression filter *normally* with data written using the direct method. They should be bit-for-bit identical I think